File: impl/app/android/app/src/main/java/com/ak/contexta/ui/components/AppBadge.kt
package com.ak.contexta.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import com.ak.contexta.ui.theme.Muted
import com.ak.contexta.ui.theme.OnPrimary
import com.ak.contexta.ui.theme.Primary
import com.ak.contexta.ui.theme.Radius
import com.ak.contexta.ui.theme.Success
import com.ak.contexta.ui.theme.SurfaceSoft

enum class AppBadgeVariant { Default, Coral, Green }

/** Small pill badge: difficulty labels, counters. */
@Composable
fun AppBadge(
    text: String,
    variant: AppBadgeVariant = AppBadgeVariant.Default,
    modifier: Modifier = Modifier
) {
    val (bg, fg) = when (variant) {
        AppBadgeVariant.Coral -> Primary to OnPrimary
        AppBadgeVariant.Green -> Success to OnPrimary
        AppBadgeVariant.Default -> SurfaceSoft to Muted
    }
    Text(
        text = text,
        style = MaterialTheme.typography.labelMedium,
        color = fg,
        modifier = modifier
            .clip(RoundedCornerShape(Radius.Pill))
            .background(bg)
            .padding(horizontal = 8.dp, vertical = 2.dp)
    )
}

File: impl/app/flutter/lib/core/components/app_badge.dart
import 'package:flutter/material.dart';

import '../theme/app_colors.dart';
import '../theme/app_dimens.dart';
import '../theme/app_type.dart';

enum AppBadgeVariant { default_, coral, green }

/// 小胶囊徽标：难度标签、计数。对照 Kotlin ui/components/AppBadge.kt。
class AppBadge extends StatelessWidget {
  const AppBadge(
    this.text, {
    super.key,
    this.variant = AppBadgeVariant.default_,
  });

  final String text;
  final AppBadgeVariant variant;

  @override
  Widget build(BuildContext context) {
    final (Color bg, Color fg) = switch (variant) {
      AppBadgeVariant.coral => (AppColors.primary, AppColors.onPrimary),
      AppBadgeVariant.green => (AppColors.success, AppColors.onPrimary),
      AppBadgeVariant.default_ => (AppColors.surfaceSoft, AppColors.muted),
    };
    return Container(
      padding: const EdgeInsets.symmetric(
        horizontal: AppSpacing.xs,
        vertical: 2,
      ),
      decoration: BoxDecoration(
        color: bg,
        borderRadius: BorderRadius.circular(AppRadius.pill),
      ),
      child: Text(
        text,
        style: AppType.textTheme.labelMedium?.copyWith(color: fg),
      ),
    );
  }
}
